### ISSUE TITLE
Test environment for baselines on Val14 and Interplan10 (DTPP)

### ISSUE THREAD

**Issue by hanbinjang** [mh0797/interPlan#7]
Hello Marcel
Thank you so much for your quick response to the previous question.

I'm evaluating DTPP and my method based on DTPP, specifically testing their performance on the Val14 and interplan benchmarks. However, there's a significant discrepancy between my experimental results and the ones reported in the paper, and I haven't been able to make any progress.

To test DTPP on Val14 and interplan, I set up the environment as follows. Could you provide some guidance or let me know if I might be missing something in my setup for testing DTPP?

Here's how I experimented:

0. **Hardware Configuration**:
   - GPU: RTX 3090, CPU: Intel i9-10980Xe

1. **Connecting tuplan_garage and DTPP**:
   - Created a YAML configuration file for DTPP:
     ```
     tuplan_garage/planning/script/config/simulation/planner/DTPP_planner.yaml
     ```

     **DTPP_planner.yaml**:
     "`yaml
     DTPP_planner:  
       _target_: tuplan_garage.planning.simulation.planner.DTPP_original.planner.Planner  
       _convert_: 'all'
     ```

   - Cloned the DTPP repository into `tuplan_garage`:
     ```
     tuplan_garage/planning/simulation/planner/DTPP_original
     ```

   - Generated a configuration file `DTPP_base_model.yaml` for the model in:
     ```
     tuplan_garage/planning/simulation/planner/DTPP_original/config
     ```

     **DTPP_base_model.yaml**:
     "`yaml
     model_path: "/model/base_model.pth"
     device: "cpu"
     ```

   - Moved `base_model.pth` to `tuplan_garage/planning/simulation/planner/DTPP_original/model`.

   - Edited `DTPP_original/planner.py` to load the base model:
     ```python
     import yaml
     import os
     current_file_path = os.path.abspath(__file__)
     current_directory = os.path.dirname(current_file_path)
     config_path = os.path.join(current_directory, 'config', 'DTPP_base_model.yaml')

     def load_config(config_path):
         with open(config_path, 'r') as f:        
             return yaml.safe_load(f)

     config = load_config(config_path)

     class Planner(AbstractPlanner):
         def __init__(self, model_path=None, device=None):        
             self._future_horizon = T  # [s]        
             self._step_interval = DT  # [s]        
             self._N_points = int(T / DT)
             self._model_path = os.path.join(current_directory, config['model_path'].lstrip('/'))
             self._device = torch.device('cpu')

         def name(self) -> str:
             return "DTPP base model"
     ```

2. **Running Simulations**:

   - **Reduced_val_14_split**:
     "`bash
     python $NUPLAN_DEVKIT_ROOT/nuplan/planning/script/run_simulation.py \
     +simulation=closed_loop_nonreactive_agents \
     planner=DTPP_planner \
     scenario_filter=reduced_val14_split \
     scenario_builder=nuplan \
     worker=ray_distributed \
     hydra.searchpath="[pkg://tuplan_garage.planning.script.config.common, pkg://tuplan_garage.planning.script.config.simulation, pkg://nuplan.planning.script.config.common, pkg://nuplan.planning.script.experiments]"
     ```

   - **interplan10**:
     "`bash
     python $INTERPLAN_ROOT/interplan/planning/script/run_simulation.py \
     +simulation=default_interplan_benchmark  \
     planner=DTPP_planner \
     scenario_filter=interplan10 \
     worker=ray_distributed \
     hydra.searchpath="[pkg://interplan.planning.script.config.common, pkg://interplan.planning.script.config.simulation, pkg://interplan.planning.script.experiments, pkg://tuplan_garage.planning.script.config.common, pkg://tuplan_garage.planning.script.config.simulation, pkg://nuplan.planning.script.config.common, pkg://nuplan.planning.script.simulation, pkg://nuplan.planning.script.experiments]"
     ```

3. **Results** (DTPP):
   - **Reduced_val_14_split**: 0.6086
   - **interplan10**: 0.3014

Could you provide any advice on this setup? Or what is your test environment for testing DTPP?

Best

Hanbin

**Comment by mh0797** [comment]
Hi @hanbinjang,
your approach looks very reasonable to me. I cannot spot any obvious bug.
When running dttp we took a slightly different approach and integrated it as a planner into tuplan_garage, i.e. we implemented the planner, respective featurebuilders etc. Similar to above, we then loaded a pretrained checkpoint to run the model.

Unfortunately, due to license issues, we cannot publish the code we used. Also, the code we reveived from the authors differs a bit from the one that was published later. In particular, the official repository contains several "bug fixes". I believe this could be the reason for the different results. 

Did you also run it on the full Val14 split?

**Comment by hanbinjang** [comment]
Thank you for the reply!

Yes, I also ran the DTPP planner on the Val14 split.

Results (DTPP):
Val14_split: 
0.7182 (Trained with validation set)
0.6591 (provided model)


Could you explain "featurebuilders" in a bit more detail?

**Comment by mh0797** [comment]
Hi,
the feature-builders are the used in the nuPlan framework to wrap the preprocessing needed to compute the features needed by the planners / models. For instance, the feature builder of PDM can be found [here](https://github.com/autonomousvision/tuplan_garage/blob/main/tuplan_garage/planning/training/preprocessing/feature_builders/pdm_feature_builder.py).

Re. the Val14 and interPlan score you get with your implementation:
I think these results look reasonable, given that their have been a few updates and bugfixes to the code in the meantime. Admittedly, it is a bit strange though that it performs worse on Val14 than the previous version. But I guess this is a bit out of scope for this repository.

**Comment by hanbinjang** [comment]
I’m truly grateful for your detailed feedback.

Thank you!
